sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -168,5 +172,83 @@ internal static string GetOperationName(Activity activity, ref AzMonList MappedT
 
             return activity.DisplayName;
         }
+
+        private static void AddExceptionTelemetryFromActivityExceptionEvents(Activity activity, ref TagEnumerationState monitorTags, string roleName, string roleInstance, string instrumentationKey, List<TelemetryItem> telemetryItems)
+        {
+            if (!activity.Events.Any())
+            {
+                return;
+            }
+
+            foreach (var evnt in activity.Events)
+            {
+                if (evnt.Name == SemanticConventions.AttributeExceptionEventName)
+                {
+                    try
+                    {
+                        var exceptionData = GetExceptionDataDetailsOnTelemetryItem(evnt.Tags);
+                        if (exceptionData != null)
+                        {
+                            var exceptionTelemetryItem = new TelemetryItem("Exception", activity, ref monitorTags, roleName, roleInstance, instrumentationKey);
+                            exceptionTelemetryItem.Data = exceptionData;
+                            telemetryItems.Add(exceptionTelemetryItem);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        AzureMonitorExporterEventSource.Log.WriteError("FailedToExtractExceptionFromActivityEvent", ex);
+                    }
+                }
+            }
+        }
+
+        internal static MonitorBase GetExceptionDataDetailsOnTelemetryItem(IEnumerable<KeyValuePair<string, object>> activityEventTags)
+        {
+            string exceptionType = null;
+            string exceptionStackTrace = null;
+            string exceptionMessage = null;
+
+            // TODO: update to use perf improvements in .NET7.0
+            foreach (var tag in activityEventTags)
+            {
+                // TODO: see if these can be cached
+                if (tag.Key == SemanticConventions.AttributeExceptionType)
+                {
+                    exceptionType = tag.Value.ToString();
+                    continue;
+                }
+                if (tag.Key == SemanticConventions.AttributeExceptionMessage)
+                {
+                    exceptionMessage = tag.Value.ToString();
+                    continue;
+                }
+                if (tag.Key == SemanticConventions.AttributeExceptionStacktrace)
+                {
+                    exceptionStackTrace = tag.Value.ToString();
+                    continue;
+                }
+            }
+
+            if (exceptionMessage == null)
+            {
+                return null;
+            }
+
+            TelemetryExceptionDetails exceptionDetails = new(exceptionMessage);
+
+            exceptionDetails.Stack = exceptionStackTrace;
+            exceptionDetails.TypeName = exceptionType;
+
+            List<TelemetryExceptionDetails> exceptions = new List<TelemetryExceptionDetails>();
```

Comment:
a new List is created just to store a single exception. Not much we can without a schema level fix. But you can create the list with size 1, to avoid wasted space even for a moment.